base/uk.ac.stfc.isis.ibex.scriptgenerator/defined_actions/action_interface.py

```diff
@@ -30,3 +33,27 @@ def parameters_valid(self, *args) -> Union[None, AnyStr]:
 
         """
         pass
+
+def cast_parameters_to(*args_casts, **keyword_casts):
+    def decorator(func):
+        @functools.wraps(func)
+        def wrapper(self, *args, **kwargs):
+            casting_failures: str = ""
+            cast_kwargs={}
+            for k, v in six.iteritems(kwargs):
```

Comment:
It may be that they only want to cast a subset of the kwargs. I propose to raise an exception only if kwargs_casts contains keys that kwargs does not?